indico/modules/events/util.py

```diff
@@ -599,13 +600,17 @@ def set_custom_fields(obj, custom_fields_data):
     return changes
 
 
-def check_permissions(event, field, allow_networks=False):
+def check_permissions(event, field, allow_networks=False, allow_registration_form=False):
     for principal_fossil, permissions in field.data:
         principal = principal_from_fossil(principal_fossil, allow_emails=True, allow_networks=allow_networks,
                                           allow_pending=True, event=event, category=event.category)
         if allow_networks and isinstance(principal, IPNetworkGroup) and set(permissions) - {READ_ACCESS_PERMISSION}:
             msg = _('IP networks cannot have management permissions: {}').format(principal.name)
             return msg
+        if (allow_registration_form and isinstance(principal, RegistrationForm)
+                and set(permissions) - {READ_ACCESS_PERMISSION}):
+            msg = _('Registration forms cannot have management permissions: {}').format(principal.name)
```

Comment:
I'd call it "Registrants" and not "Registration forms". While the object you reference is a registration form, the users in one are registrants.